uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/app/App.java

```diff
@@ -101,8 +101,16 @@ public void install() {
         // the app is just installed, first run should be handled
         // this value can (should) be changed to false by child classes as appropriate
         shouldHandleFirstRun = true;
+
+        // setting app implementation for the installed app based on version
+        initialiseAppImpl();
     }
 
+    /**
+     * Initialise implementation of app based on version
+     */
+    abstract protected void initialiseAppImpl();
```

Comment:
Thinking about it more, now that since we're not doing it with generics, I think this leading to the following code in a lot of places:

```java
 @Override
    public void initialiseAppImpl() {
       // nothing required
    }
```

I think this is going to cause some confusion for folks reading the code. I suggest that we either remove it from `App.java` and move it only to the authenticator class or if we want to keep it on `App.java` then we can provide a default implementation that sets `mAppImpl` to `this`

Reply:
If we go with first approach ( I suggest that we either remove it from App.java and move it only to the authenticator class), then this is not scalable to support multiple versions for all apps. 
I think the second approach is what I tried to do earlier, but the problem is mAppImpl is not defined in App.java. It is defined in each of the app 's child classes. So, I went with the approach of having private impl variable defined in each class and in each class, when we want to support running on multiple app versions, we can remove the //nothing required here comment and add the relevant code. If "nothing required here" is causing the confusion, I can add a To-Do in each class to say that when we want to support multiple versions, we can add it in this function (initialiseAppImpl)